libs/SmartSync/SmartSync/Classes/Manager/SFSmartSyncSyncManager.m

```diff
@@ -107,19 +107,32 @@ + (void)removeSharedInstance:(SFUserAccount*)user {
 + (void)removeSharedInstanceForUser:(SFUserAccount *)user storeName:(NSString *)storeName {
     if (user == nil) return;
     if (storeName.length == 0) storeName = kDefaultSmartStoreName;
-    SFSmartStore *store = [SFSmartStore sharedStoreWithName:storeName user:user];
-    [self removeSharedInstanceForStore:store];
+    NSString* key = [SFSmartSyncSyncManager keyForUser:user storeName:storeName];
+    [SFSmartSyncSyncManager removeSharedInstanceForKey:key];
+}
+
++ (void)removeSharedInstanceForStore:(SFSmartStore*) store {
+    NSString* key = [SFSmartSyncSyncManager keyForStore:store];
+    [SFSmartSyncSyncManager removeSharedInstanceForKey:key];
 }
 
-+ (void)removeSharedInstanceForStore:(SFSmartStore *)store {
++ (void)removeSharedInstanceForKey:(NSString*) key {
     @synchronized([SFSmartSyncSyncManager class]) {
-        if (store.storePath.length > 0) {
-            NSString *key = store.storePath;
-            [syncMgrList removeObjectForKey:key];
-        }
+        [syncMgrList removeObjectForKey:key];
     }
 }
 
++ (NSString*)keyForStore:(SFSmartStore*)store {
+    return [SFSmartSyncSyncManager keyForUser:store.user storeName:store.storeName];
+}
+
++ (NSString*)keyForUser:(SFUserAccount*)user storeName:(NSString*)storeName {
+    NSString* keyPrefix = user == nil ? SFKeyForUserAndScope(nil, SFUserAccountScopeGlobal) : SFKeyForUserAndScope(user, SFUserAccountScopeUser);
```

Comment:
This should default to ```SFUserAccountScopeCommunity``` I believe. In the absence of community, it will use org + user. User alone is inadequate for uniqueness.